src/Microsoft.Health.Fhir.SqlServer/Features/Schema/SchemaInitializer.cs

```diff
@@ -0,0 +1,143 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Data.SqlClient;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.SqlServer.Configs;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Schema
+{
+    /// <summary>
+    /// EXPERIMENTAL - Initializes the sql schema and brings the schema up to the min supported version.
+    /// The purpose of this it to enable easy scenarios during development and will likely be removed later.
+    /// </summary>
+    public class SchemaInitializer : IStartable
+    {
+        private readonly SqlServerDataStoreConfiguration _sqlServerDataStoreConfiguration;
+        private readonly SchemaUpgradeRunner _schemaUpgradeRunner;
+        private readonly SchemaInformation _schemaInformation;
+        private readonly ILogger<SchemaInitializer> _logger;
+
+        public SchemaInitializer(SqlServerDataStoreConfiguration sqlServerDataStoreConfiguration, SchemaUpgradeRunner schemaUpgradeRunner, SchemaInformation schemaInformation, ILogger<SchemaInitializer> logger)
+        {
+            EnsureArg.IsNotNull(sqlServerDataStoreConfiguration, nameof(sqlServerDataStoreConfiguration));
+            EnsureArg.IsNotNull(schemaUpgradeRunner, nameof(schemaUpgradeRunner));
+            EnsureArg.IsNotNull(schemaInformation, nameof(schemaInformation));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _sqlServerDataStoreConfiguration = sqlServerDataStoreConfiguration;
+            _schemaUpgradeRunner = schemaUpgradeRunner;
+            _schemaInformation = schemaInformation;
+            _logger = logger;
+        }
+
+        private void Initialize()
+        {
+            if (!CanInitialize())
+            {
+                return;
+            }
+
+            if (!IsSchemaTableInitialized())
+            {
+                CreateSchemaTable();
+            }
+
+            SetCurrentSchemaVersion();
+
+            if (_schemaInformation.Current == null || _schemaInformation.Current < _schemaInformation.Min)
+            {
+                int current = (int?)_schemaInformation.Current ?? 0;
+
+                for (int i = current + 1; i <= (int)_schemaInformation.Min; i++)
```

Comment:
Max?

Reply:
My thought was that it's more likely when you're first booting it up that you'd want to test upgrades.

Reply:
And then how do you apply the lastest version?

Reply:
Updating to do Max version since that's likely the more common scenario.